libs/oidc/provider/src/lib/controllers/interaction/interaction.controller.ts

```diff
@@ -136,7 +141,16 @@ export class InteractionController {
     if (locals.method) {
       if (locals.method === 'totp') {
         const token = otplib.totp.generate(user.secret2fa);
-        Mailer.sendTokenByEmail(user, token);
+
+        this.mailerService.sendMail({
+          to: user.email,
+          from: '"GISC Accounts Team" <giscaccounts@tamu.edu>',
+          subject: 'Hello from the IdP',
+          text: `Your two-step verification code is: ${token}`,
+          html:
+            `<p>Your two-step verification code is: <b>${token}</b></p>` +
+            `<p>Use this code to complete loggin in with GISC</p>`
+        });
```

Comment:
Should it be awaited? All of these emails are being sent out of order and CAN result in conditions where operations before or after the mail send fail and will either send an email regardless or not send one at all. Example:

1) View for 2fa confirmation gets rendered
2) Email with 2fa code doesn't get sent because of an error in 2fa code generation or email sending itself.